include/aws/common/file.h

```diff
@@ -4,7 +4,58 @@
  * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0.
  */
+#include <aws/common/byte_buf.h>
 #include <aws/common/common.h>
+#include <aws/common/linked_list.h>
+#include <aws/common/ref_count.h>
+
+#ifdef _WIN32
+static const char AWS_PATH_DELIM = '\\';
+#else
+static const char AWS_PATH_DELIM = '/';
+#endif
+
+enum aws_file_type {
+    AWS_FILE_TYPE_NONE,
```

Comment:
The way this is used, it's more like UNKNOWN, isnt it?

Reply:
Well, for one, the info needs to be stored on the object. lstat is expensive and I don't wanna call it on demand, but I could do a bit mask and add helper functions.